File: lesson6/nestedTable.js
function helper(acc, idx, n) {
  if (idx === n) {
    const array = new Array(
    return;
  }


}

function createNestedArray(numbers) {
  let accumulator = []


  let nestedArray = [];
  let currentArray = nestedArray;

  for (let i = 0; i < numbers.length; i++) {
    // let num = numbers[i];

    // for (let j = 0; j < num; j++) {
    //   if (currentArray[j] === undefined) {
    //     currentArray[j] = [];
    //   }

    //   currentArray = currentArray[j];
    // }

    // currentArray = nestedArray;
  }

  return nestedArray;
}

console.log(createNestedArray([2, 3, 1]))

File: lesson6/nestedTable.lua
local pt    = require("pt")

function createNestedTable(numbers)
  local nestedTable = {}
  local currentTable = nestedTable
  
  for i = 1, #numbers do
    local num = numbers[i]
    
    for j = 1, num do
      if j == num then
        currentTable[j] = {}
        currentTable = currentTable[j]
      else
        if not currentTable[j] then
          currentTable[j] = {}
        end
        
        currentTable = currentTable[j]
      end
    end
    
    currentTable = nestedTable
  end
  
  return nestedTable
end

print(pt.pt(createNestedTable({2, 3, 1})))
